packages/x-data-grid/src/material/index.tsx

```diff
@@ -340,8 +331,32 @@ function BaseAutocomplete(props: GridSlotProps['baseAutocomplete']) {
 }
 
 function BaseInput(props: GridSlotProps['baseInput']) {
+  return <MUIInputBase {...transformInputProps(props)} />;
+}
+
+function transformInputProps(props: GridSlotProps['baseInput'] | undefined) {
+  if (!props) {
+    return undefined;
+  }
+
   const { slotProps, ...rest } = props;
-  return <MUIInputBase {...rest} inputProps={slotProps?.htmlInput} />;
+  const result = rest as Partial<MUIInputBaseProps>;
+
+  if (result.startAdornment) {
+    result.startAdornment = (
+      <MUIInputAdornment position="start">{result.startAdornment}</MUIInputAdornment>
+    );
+  }
+
+  if (result.endAdornment) {
+    result.endAdornment = (
+      <MUIInputAdornment position="end">{result.endAdornment}</MUIInputAdornment>
```

Comment:
I added a style override to icon buttons in input adornments here https://github.com/mui/mui-x/pull/16630

When you've synced master, it's fine to remove the `BaseInputAdornment` I added and add the style override here

```suggestion
      <MUIInputAdornment position="end" sx={{ [`& .${iconButtonClasses.sizeSmall}`]: { marginRight: '-7px' } }}>{result.endAdornment}</MUIInputAdornment>
```